modules/sc-mesh-secure-deployment/src/nats/src/bat_ctrl_utils.py

```diff
@@ -0,0 +1,280 @@
+            )
+
+            mtu_size = "1500" if is_upper else "1546"
+            subprocess.run(
+                ["ip", "link", "set", "dev", batman_if, "mtu", mtu_size], check=True
+            )
+        except subprocess.CalledProcessError as e:
+            self.logger.error("Error configuring BATMAN interface: %s", e)
+
+    def set_hop_penalty(self) -> None:
+        """
+        Sets Batman hop penalty for interfaces defined in config file.
+        """
+        if self.__hop_penalty is None:
+            return
+        # Set hop penalty for mesh interfaces
+        meshif_hop_penalty = self.__hop_penalty.get("meshif", {})
+        self.logger.info(f"meshif_hop_penalty: {meshif_hop_penalty}")
+        if meshif_hop_penalty is not None:
+            for interface, penalty in meshif_hop_penalty.items():
+                if penalty is not None:
+                    try:
+                        subprocess.run(
+                            ["batctl", "meshif", interface, "hop_penalty", str(penalty)],
+                            check=True,
+                        )
+                    except Exception as e:
+                        self.logger.info(
+                            "Failed to set hop penalty %s for: %s. Error: %s",
+                            penalty,
+                            interface,
+                            e,
+                        )
+
+        # Set hop penalty for hard interfaces
+        hardif_hop_penalty = self.__hop_penalty.get("hardif", {})
+        self.logger.info(f"hardif_hop_penalty: {hardif_hop_penalty}")
+        if hardif_hop_penalty is not None:
+            for interface, penalty in hardif_hop_penalty.items():
+                if penalty is not None:
+                    try:
+                        # Serch hardif from all my batmans
+                        for batman_interface in self.__bat_interfaces:
+                            hardif = self.__find_batman_hardif(interface, batman_interface)
+                            if hardif:
+                                subprocess.run(
+                                    [
+                                        "batctl",
+                                        "hardif",
+                                        hardif,
+                                        "hop_penalty",
+                                        str(penalty),
+                                    ],
+                                    check=True,
+                                )
+                                break
+                    except Exception as e:
+                        self.logger.info(
+                            "Failed to set hop penalty %s for: %s. Error: %s",
+                            penalty,
+                            interface,
+                            e,
+                        )
+
+    @staticmethod
+    def __get_interface_mac(interface: str) -> Union[None, str]:
```

Comment:
Maybe not bat_ctrl related functionality.  Probably we need network_utils also for common networking methods.

Reply:
No need to fix it now.